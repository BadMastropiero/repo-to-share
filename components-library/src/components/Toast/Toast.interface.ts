import {FC} from 'react';
import {CustomIconProps} from '../Icon/Icon.interface';

export interface ToastProps {
  message?: string;
  description?: string;
  actionLabel?: string;
  noIcon?: boolean;
  noCloseButton?: boolean;
  noAction?: boolean;
  onActionClicked?: () => void;
  onDismissed?: () => void;
  isOpen?: boolean;
  setIsOpen: (val: boolean) => void;
  background?: string;
  highlightColor?: string;
  icon?: FC<CustomIconProps>;
  type?: 'error' | 'warning' | 'success' | 'info' | 'default';
  timeout?: number;
  onTimeout?: () => void;
  isSnackbar?: boolean;
  className?: string;
  style?: React.CSSProperties | undefined;
}
