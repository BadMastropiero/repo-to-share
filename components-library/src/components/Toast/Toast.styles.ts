import styled, {css} from 'styled-components';
import {Button} from '../Button';
import {Icon} from '../Icon';

const StyledToast = styled.div<{background?: string; highlightColor?: string}>(
  ({
    theme: {
      colors: {primaryBg, primaryLight},
    },
    background = primaryLight,
  }) => css`
    display: flex;
    flex-shrink: 0;
    flex-grow: 0;
    flex-direction: row;
    align-items: flex-start;
    border: 4px solid ${primaryBg};
    padding: 16px;
    gap: 20px;
    border-radius: 10px;
    background: ${background};
    box-shadow: 0px 10px 15px -5px rgba(0, 101, 166, 0.1);
    word-wrap: break-word;
  `
);

export const ToastContent = styled.div`
  flex: 2;
  min-width: min-content;
  display: flex;
  flex-shrink: 0;
  flex-direction: column;
  justify-items: flex-start;
  row-gap: 10px;
  word-break: break-word;
`;

export const ToastHeader = styled.div<{isSnackbar?: boolean}>(
  ({isSnackbar = false}) => css`
    display: flex;
    flex-shrink: 0;
    flex-direction: row;
    column-gap: 16px;
    justify-content: space-between;
    align-items: ${isSnackbar ? 'center' : 'flex-start'};
    max-width: 100%;
  `
);

export const ToastButtonsContainer = styled.div`
  display: flex;
  flex-shrink: 0;
  flex-direction: row;
  column-gap: 16px;
  justify-content: flex-end;
  align-items: center;
`;

export const ToastButton = styled(Button)<{highlightColor?: string; hoverColor?: string}>(
  ({highlightColor = 'white', hoverColor}) => css`
    border: unset;
    background: transparent;
    box-shadow: unset;
    flex-shrink: 0;

    &.default {
      :hover:not(:disabled) {
        p:first-child {
          color: ${hoverColor};
          svg:first-child {
            fill: ${hoverColor};
          }
        }

        div:first-child {
          color: ${hoverColor};
          svg:first-child {
            fill: ${hoverColor};
          }
        }
      }
    }

    :hover:not(:disabled) {
      div:first-child {
        background: white;
        svg:first-child {
          fill: ${highlightColor};
        }
      }

      p:first-child {
        background: white;
        svg:first-child {
          fill: ${highlightColor};
        }
      }
    }

    p:first-child {
      font-weight: 500;
      font-size: 14px;
      line-height: 16.4px;
      background: transparent;
      color: ${highlightColor};
    }

    div:first-child {
      font-weight: 500;
      font-size: 14px;
      line-height: 16.4px;
      background: transparent;
      color: ${highlightColor};

      svg:first-child {
        height: 14px; // should match action text size
        width: 14px;
        padding: 2px;
        fill: ${highlightColor};
      }
    }
  `
);

export const ToastIcon = styled(Icon)<{correctAlign?: boolean; isSnackbar?: boolean}>(
  ({correctAlign = false, isSnackbar = false}) => css`
    flex-shrink: 0;
    align-self: center;
    // vertical alignment of icon with title header necessary when toast
    margin-top: ${(isSnackbar && '0px') || (correctAlign ? '-3px' : '2px')};
  `
);

export default StyledToast;
