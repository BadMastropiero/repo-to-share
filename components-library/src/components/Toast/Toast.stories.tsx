import {useEffect, useState} from 'react';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import {Button} from '../Button';
import Toast from './Toast';
import {Typography} from '../Typography';

export default {
  title: 'Components/Toast',
  component: Toast,
} as ComponentMeta<typeof Toast>;

export const Base: ComponentStory<typeof Toast> = args => {
  const {
    isOpen: visible = false,
    actionLabel,
    background,
    highlightColor,
    message: title,
    description: message,
    noCloseButton,
    noAction,
    noIcon,
    type,
    timeout,
    isSnackbar,
  } = args;

  const [isVisible, setIsVisible] = useState<boolean>(visible);
  const [handledText, setHandledText] = useState<string>(
    'Interact with the toast to see it results'
  );

  useEffect(() => {
    setIsVisible(visible);
  }, [visible]);

  const launchToast = () => {
    setIsVisible(true);
  };

  const handleAction = () => {
    setHandledText("You've triggered the action");
  };

  const handleDismiss = () => {
    setHandledText("You've dismissed");
  };

  return (
    <>
      <Button onClick={launchToast} presetSize="small" text="Launch Toast" primary rounded />
      <Typography padding="10px">{handledText}</Typography>
      <Toast
        message={title}
        description={message}
        actionLabel={actionLabel}
        onActionClicked={handleAction}
        onDismissed={handleDismiss}
        isOpen={isVisible}
        setIsOpen={setIsVisible} // expose this prop
        background={background}
        highlightColor={highlightColor}
        noCloseButton={noCloseButton}
        noIcon={noIcon}
        noAction={noAction}
        type={type}
        timeout={timeout}
        isSnackbar={isSnackbar}
      />
    </>
  );
};

Base.args = {
  message: 'Toast Title',
  description: 'Toast message: Lorem ipsum',
  isOpen: false,
  background: undefined,
  highlightColor: undefined,
  actionLabel: undefined,
  noIcon: false,
  noAction: false,
  noCloseButton: false,
  type: undefined,
  timeout: undefined,
  isSnackbar: undefined,
};

Base.parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/epHXucNHjke2rES69iOsaS/TetraImpacts?node-id=2213%3A4529&t=zcHB2Up79k7uM2aA-4',
  },
};
