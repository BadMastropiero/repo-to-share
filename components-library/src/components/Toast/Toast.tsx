import React, {useContext, useEffect} from 'react';
import {ThemeContext} from 'styled-components';

// import {Button} from '../Button';
import {Close, Error, Info, Task, Warning} from '../Icon';
import {Typography} from '../Typography';

// Interfaces
import {ToastProps} from './Toast.interface';

// Styles
import StyledToast, {
  ToastButtonsContainer,
  ToastButton,
  ToastContent,
  ToastHeader,
  ToastIcon,
} from './Toast.styles';

const Toast: React.FC<ToastProps> = ({
  message = 'Message',
  description = 'Description',
  actionLabel = 'Action',
  onActionClicked,
  onDismissed,
  onTimeout = () => null,
  noIcon,
  noCloseButton = false,
  noAction = false,
  isOpen = false,
  setIsOpen,
  background,
  highlightColor = '#FFFFFF',
  icon,
  type,
  timeout = 5000,
  isSnackbar = false,
  className,
  style,
}) => {
  const {colors} = useContext(ThemeContext);

  const presetBackground = type ? colors.toast[type.concat('Bg')] : background;
  const presetHighlight = type ? colors.toast[type] : highlightColor;

  const baseClassName = `ti-${isSnackbar ? 'snackbar' : 'toast'}`;

  const mapTypeToIcon = {
    error: Error,
    info: Info,
    success: Task,
    warning: Warning,
    default: Info,
  };

  const handleDismiss = () => {
    if (onDismissed) {
      onDismissed();
    }
    setIsOpen(false);
  };

  const handleAction = () => {
    if (onActionClicked) {
      onActionClicked();
    }
    setIsOpen(false);
  };

  useEffect(() => {
    const tim = setTimeout(() => {
      if (isOpen) {
        onTimeout();
        setIsOpen(false);
      }
    }, timeout);
    return () => {
      clearTimeout(tim);
    };
  }, [isOpen]);

  return isOpen ? (
    <StyledToast
      className={`${baseClassName} ${className}`}
      background={presetBackground}
      highlightColor={presetHighlight}
      style={style}
    >
      {noIcon || (
        <ToastIcon
          isSnackbar={isSnackbar}
          icon={icon || mapTypeToIcon[type || 'default']}
          customColor={presetHighlight}
          primary
          correctAlign={noAction && noCloseButton} // correct the alignment taking into account action or close button existence
        />
      )}
      <ToastContent className={`${baseClassName}-content`}>
        <ToastHeader isSnackbar={isSnackbar} className={`${baseClassName}-header`}>
          <Typography
            fontWeight={isSnackbar ? '500' : '700'}
            fontSize={isSnackbar ? '14px' : '16px'}
            height={isSnackbar ? '16.4px' : '19px'}
            lineHeight={isSnackbar ? '16.4px' : '19px'}
            fontFamily="Roboto Flex"
            fontStyle="normal"
            colorPreset="primary"
            style={{
              marginTop: `${
                isSnackbar
                  ? (noAction && noCloseButton && '3px') || '0px'
                  : (noAction && noCloseButton && '0px') || '5px'
              }`,
            }}
            customColor={presetHighlight}
          >
            {message}
          </Typography>
          {(noCloseButton && noAction) || (
            <ToastButtonsContainer className={`${baseClassName}-container`}>
              {noAction || (
                <ToastButton
                  className={`${baseClassName}-action-btn ${type || 'default'}`}
                  presetSize="small"
                  text={actionLabel}
                  rounded
                  onClick={handleAction}
                  highlightColor={type ? presetHighlight : colors?.toast?.default}
                  hoverColor={type ? presetBackground : colors?.toast?.defaultBg}
                />
              )}
              {noCloseButton || (
                <ToastButton
                  className={`${baseClassName}-close-btn ${type || 'default'}`}
                  presetSize="small"
                  afterIcon={Close}
                  rounded
                  onClick={handleDismiss}
                  highlightColor={type ? presetHighlight : colors?.toast?.default}
                  hoverColor={type ? presetBackground : colors?.toast?.defaultBg}
                />
              )}
            </ToastButtonsContainer>
          )}
        </ToastHeader>
        {isSnackbar || (
          <Typography
            fontWeight="500"
            fontSize="12px"
            lineHeight="20.4px"
            fontFamily="Roboto Flex"
            fontStyle="normal"
            colorPreset="disable"
            customColor={presetHighlight}
          >
            {description}
          </Typography>
        )}
      </ToastContent>
    </StyledToast>
  ) : null;
};

export default Toast;
