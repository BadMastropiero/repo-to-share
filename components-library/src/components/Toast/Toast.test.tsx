import {render, waitForElementToBeRemoved} from '@testing-library/react';
import {composeStories} from '@storybook/testing-react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import * as stories from './Toast.stories';

const {Base} = composeStories(stories);

describe('Toast test', () => {
  test('should match snapshot', () => {
    const tree = render(<Base isOpen />);
    expect(tree).toMatchSnapshot();
  });

  test('should appear, then disappear execute `ACTION` button', () => {
    const tree = render(<Base />);
    const item = tree.getByText('Launch Toast');

    userEvent.click(item);

    const ButtonInsideCard = tree.getByText('Action').parentElement;
    expect(ButtonInsideCard).toBeInTheDocument();

    if (ButtonInsideCard) {
      userEvent.click(ButtonInsideCard);
      expect(ButtonInsideCard).not.toBeInTheDocument();
    }
  });

  test('should appear, then automatically disappear', async () => {
    const tree = render(<Base timeout={1000} />);
    const item = tree.getByText('Launch Toast');

    userEvent.click(item);

    const ButtonInsideCard = tree.getByText('Action').parentElement;
    expect(ButtonInsideCard).toBeInTheDocument();
    await waitForElementToBeRemoved(() => tree.queryByText('Action'));
  });
});
