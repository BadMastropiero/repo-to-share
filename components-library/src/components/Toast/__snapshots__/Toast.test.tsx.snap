// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast test should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-bcXHqe sc-dkrFOg gkrZeY ti-button "
      >
        <p
          class="sc-bcXHqe sc-eDvSVe Hvoxq"
        >
          Launch Toast
        </p>
        <div
          class="sc-gKPRtg dOMrEj"
          color="#C5CFDF"
        />
      </button>
      <div
        class="sc-bcXHqe sc-eDvSVe dfLAUW"
      >
        Interact with the toast to see it results
      </div>
      <div
        class="sc-pyfCe kHHhkK ti-toast undefined"
      >
        <svg
          class="sc-jSUZER rbNgI ticons-info sc-eDWCr dkwEuI"
          color="#FFFFFF"
          height="24px"
          role="img"
          viewBox="0 0 24 24"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.025 17.4C12.3916 17.4 12.7043 17.2706 12.963 17.012C13.221 16.754 13.35 16.4416 13.35 16.075V12.3C13.35 11.9333 13.221 11.625 12.963 11.375C12.7043 11.125 12.3916 11 12.025 11C11.6583 11 11.346 11.129 11.088 11.387C10.8293 11.6456 10.7 11.9583 10.7 12.325V16.1C10.7 16.4666 10.8293 16.775 11.088 17.025C11.346 17.275 11.6583 17.4 12.025 17.4ZM12 8.59995C12.35 8.59995 12.6456 8.47928 12.887 8.23795C13.129 7.99595 13.25 7.69995 13.25 7.34995C13.25 6.99995 13.129 6.70395 12.887 6.46195C12.6456 6.22062 12.35 6.09995 12 6.09995C11.65 6.09995 11.3543 6.22062 11.113 6.46195C10.871 6.70395 10.75 6.99995 10.75 7.34995C10.75 7.69995 10.871 7.99595 11.113 8.23795C11.3543 8.47928 11.65 8.59995 12 8.59995ZM12 22.7999C10.5 22.7999 9.09562 22.5166 7.78695 21.95C6.47895 21.3833 5.33762 20.6126 4.36295 19.638C3.38762 18.6626 2.61662 17.5206 2.04995 16.212C1.48328 14.904 1.19995 13.5 1.19995 12C1.19995 10.5 1.48328 9.09162 2.04995 7.77495C2.61662 6.45828 3.38762 5.31262 4.36295 4.33795C5.33762 3.36262 6.47895 2.59562 7.78695 2.03695C9.09562 1.47895 10.5 1.19995 12 1.19995C13.5 1.19995 14.9083 1.47895 16.225 2.03695C17.5416 2.59562 18.6876 3.36262 19.663 4.33795C20.6376 5.31262 21.4039 6.45828 21.962 7.77495C22.5206 9.09162 22.7999 10.5 22.7999 12C22.7999 13.5 22.5206 14.904 21.962 16.212C21.4039 17.5206 20.6376 18.6626 19.663 19.638C18.6876 20.6126 17.5416 21.3833 16.225 21.95C14.9083 22.5166 13.5 22.7999 12 22.7999ZM12 20.15C14.2666 20.15 16.1916 19.3583 17.775 17.775C19.3583 16.1916 20.15 14.2666 20.15 12C20.15 9.73328 19.3583 7.80828 17.775 6.22495C16.1916 4.64162 14.2666 3.84995 12 3.84995C9.73328 3.84995 7.80828 4.64162 6.22495 6.22495C4.64162 7.80828 3.84995 9.73328 3.84995 12C3.84995 14.2666 4.64162 16.1916 6.22495 17.775C7.80828 19.3583 9.73328 20.15 12 20.15Z"
          />
        </svg>
        <div
          class="sc-jrcTuL hNCoUk ti-toast-content"
        >
          <div
            class="sc-kDvujY ewIPOD ti-toast-header"
          >
            <div
              class="sc-bcXHqe sc-eDvSVe jOVTOx"
              font-family="Roboto Flex"
              font-size="16px"
              font-style="normal"
              font-weight="700"
              height="19px"
              style="margin-top: 5px;"
            >
              Toast Title
            </div>
            <div
              class="sc-ipEyDJ ippIiE ti-toast-container"
            >
              <button
                class="sc-bcXHqe sc-dkrFOg iLwNZv ti-button sc-csuSiG isuCcs ti-toast-action-btn default"
              >
                <p
                  class="sc-bcXHqe sc-eDvSVe kEAGbr"
                >
                  Action
                </p>
                <div
                  class="sc-gKPRtg bJoWGa"
                  color="#FFFFFF"
                />
              </button>
              <button
                class="sc-bcXHqe sc-dkrFOg iLwNZv ti-button sc-csuSiG isuCcs ti-toast-close-btn default"
              >
                <div
                  class="sc-gswNZR eYhtjX"
                >
                  <svg
                    class="sc-jSUZER hAlpLQ ticons-close undefined"
                    color="#0065A6"
                    height="16px"
                    role="img"
                    viewBox="0 0 24 24"
                    width="16px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.05 13.85L7.27495 18.625C7.00828 18.8917 6.69162 19.025 6.32495 19.025C5.95828 19.025 5.64162 18.8917 5.37495 18.625C5.10828 18.3584 4.97495 18.046 4.97495 17.688C4.97495 17.3294 5.10828 17.0167 5.37495 16.75L10.175 11.95L5.34995 7.15002C5.08328 6.88336 4.94995 6.57502 4.94995 6.22502C4.94995 5.87502 5.08328 5.56669 5.34995 5.30002C5.61662 5.03336 5.92895 4.90002 6.28695 4.90002C6.64562 4.90002 6.95828 5.03336 7.22495 5.30002L12 10.05L16.725 5.32502C16.9916 5.05836 17.3083 4.92502 17.675 4.92502C18.0416 4.92502 18.3583 5.05836 18.625 5.32502C18.8916 5.59169 19.025 5.91236 19.025 6.28702C19.025 6.66236 18.8916 6.98336 18.625 7.25002L13.875 12.025L18.65 16.8C18.9 17.05 19.025 17.3584 19.025 17.725C19.025 18.0917 18.9 18.4 18.65 18.65C18.4 18.9 18.096 19.025 17.738 19.025C17.3793 19.025 17.075 18.9 16.825 18.65L12.05 13.85Z"
                    />
                  </svg>
                </div>
                <div
                  class="sc-gKPRtg bJoWGa"
                  color="#FFFFFF"
                />
              </button>
            </div>
          </div>
          <div
            class="sc-bcXHqe sc-eDvSVe jxPpwj"
            font-family="Roboto Flex"
            font-size="12px"
            font-style="normal"
            font-weight="500"
          >
            Toast message: Lorem ipsum
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-bcXHqe sc-dkrFOg gkrZeY ti-button "
    >
      <p
        class="sc-bcXHqe sc-eDvSVe Hvoxq"
      >
        Launch Toast
      </p>
      <div
        class="sc-gKPRtg dOMrEj"
        color="#C5CFDF"
      />
    </button>
    <div
      class="sc-bcXHqe sc-eDvSVe dfLAUW"
    >
      Interact with the toast to see it results
    </div>
    <div
      class="sc-pyfCe kHHhkK ti-toast undefined"
    >
      <svg
        class="sc-jSUZER rbNgI ticons-info sc-eDWCr dkwEuI"
        color="#FFFFFF"
        height="24px"
        role="img"
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.025 17.4C12.3916 17.4 12.7043 17.2706 12.963 17.012C13.221 16.754 13.35 16.4416 13.35 16.075V12.3C13.35 11.9333 13.221 11.625 12.963 11.375C12.7043 11.125 12.3916 11 12.025 11C11.6583 11 11.346 11.129 11.088 11.387C10.8293 11.6456 10.7 11.9583 10.7 12.325V16.1C10.7 16.4666 10.8293 16.775 11.088 17.025C11.346 17.275 11.6583 17.4 12.025 17.4ZM12 8.59995C12.35 8.59995 12.6456 8.47928 12.887 8.23795C13.129 7.99595 13.25 7.69995 13.25 7.34995C13.25 6.99995 13.129 6.70395 12.887 6.46195C12.6456 6.22062 12.35 6.09995 12 6.09995C11.65 6.09995 11.3543 6.22062 11.113 6.46195C10.871 6.70395 10.75 6.99995 10.75 7.34995C10.75 7.69995 10.871 7.99595 11.113 8.23795C11.3543 8.47928 11.65 8.59995 12 8.59995ZM12 22.7999C10.5 22.7999 9.09562 22.5166 7.78695 21.95C6.47895 21.3833 5.33762 20.6126 4.36295 19.638C3.38762 18.6626 2.61662 17.5206 2.04995 16.212C1.48328 14.904 1.19995 13.5 1.19995 12C1.19995 10.5 1.48328 9.09162 2.04995 7.77495C2.61662 6.45828 3.38762 5.31262 4.36295 4.33795C5.33762 3.36262 6.47895 2.59562 7.78695 2.03695C9.09562 1.47895 10.5 1.19995 12 1.19995C13.5 1.19995 14.9083 1.47895 16.225 2.03695C17.5416 2.59562 18.6876 3.36262 19.663 4.33795C20.6376 5.31262 21.4039 6.45828 21.962 7.77495C22.5206 9.09162 22.7999 10.5 22.7999 12C22.7999 13.5 22.5206 14.904 21.962 16.212C21.4039 17.5206 20.6376 18.6626 19.663 19.638C18.6876 20.6126 17.5416 21.3833 16.225 21.95C14.9083 22.5166 13.5 22.7999 12 22.7999ZM12 20.15C14.2666 20.15 16.1916 19.3583 17.775 17.775C19.3583 16.1916 20.15 14.2666 20.15 12C20.15 9.73328 19.3583 7.80828 17.775 6.22495C16.1916 4.64162 14.2666 3.84995 12 3.84995C9.73328 3.84995 7.80828 4.64162 6.22495 6.22495C4.64162 7.80828 3.84995 9.73328 3.84995 12C3.84995 14.2666 4.64162 16.1916 6.22495 17.775C7.80828 19.3583 9.73328 20.15 12 20.15Z"
        />
      </svg>
      <div
        class="sc-jrcTuL hNCoUk ti-toast-content"
      >
        <div
          class="sc-kDvujY ewIPOD ti-toast-header"
        >
          <div
            class="sc-bcXHqe sc-eDvSVe jOVTOx"
            font-family="Roboto Flex"
            font-size="16px"
            font-style="normal"
            font-weight="700"
            height="19px"
            style="margin-top: 5px;"
          >
            Toast Title
          </div>
          <div
            class="sc-ipEyDJ ippIiE ti-toast-container"
          >
            <button
              class="sc-bcXHqe sc-dkrFOg iLwNZv ti-button sc-csuSiG isuCcs ti-toast-action-btn default"
            >
              <p
                class="sc-bcXHqe sc-eDvSVe kEAGbr"
              >
                Action
              </p>
              <div
                class="sc-gKPRtg bJoWGa"
                color="#FFFFFF"
              />
            </button>
            <button
              class="sc-bcXHqe sc-dkrFOg iLwNZv ti-button sc-csuSiG isuCcs ti-toast-close-btn default"
            >
              <div
                class="sc-gswNZR eYhtjX"
              >
                <svg
                  class="sc-jSUZER hAlpLQ ticons-close undefined"
                  color="#0065A6"
                  height="16px"
                  role="img"
                  viewBox="0 0 24 24"
                  width="16px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.05 13.85L7.27495 18.625C7.00828 18.8917 6.69162 19.025 6.32495 19.025C5.95828 19.025 5.64162 18.8917 5.37495 18.625C5.10828 18.3584 4.97495 18.046 4.97495 17.688C4.97495 17.3294 5.10828 17.0167 5.37495 16.75L10.175 11.95L5.34995 7.15002C5.08328 6.88336 4.94995 6.57502 4.94995 6.22502C4.94995 5.87502 5.08328 5.56669 5.34995 5.30002C5.61662 5.03336 5.92895 4.90002 6.28695 4.90002C6.64562 4.90002 6.95828 5.03336 7.22495 5.30002L12 10.05L16.725 5.32502C16.9916 5.05836 17.3083 4.92502 17.675 4.92502C18.0416 4.92502 18.3583 5.05836 18.625 5.32502C18.8916 5.59169 19.025 5.91236 19.025 6.28702C19.025 6.66236 18.8916 6.98336 18.625 7.25002L13.875 12.025L18.65 16.8C18.9 17.05 19.025 17.3584 19.025 17.725C19.025 18.0917 18.9 18.4 18.65 18.65C18.4 18.9 18.096 19.025 17.738 19.025C17.3793 19.025 17.075 18.9 16.825 18.65L12.05 13.85Z"
                  />
                </svg>
              </div>
              <div
                class="sc-gKPRtg bJoWGa"
                color="#FFFFFF"
              />
            </button>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-eDvSVe jxPpwj"
          font-family="Roboto Flex"
          font-size="12px"
          font-style="normal"
          font-weight="500"
        >
          Toast message: Lorem ipsum
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
